packages/mobile/src/components/add-to-collection-drawer/AddToCollectionDrawer.tsx

```diff
@@ -66,53 +81,28 @@ export const AddToCollectionDrawer = () => {
   const trackId = useSelector(getTrackId)
   const trackTitle = useSelector(getTrackTitle)
   const isTrackUnlisted = useSelector(getTrackIsUnlisted)
-  const account = useSelector(getAccountWithNameSortedPlaylistsAndAlbums)
-  const currentCollectionId = useSelector(getCollectionId)
+  const [filter, setFilter] = useState('')
 
   const messages = getMessages(collectionType)
 
   useEffectOnce(() => {
     dispatch(fetchAccountCollections())
   })
 
-  const renderImage = useCallback(
-    (item) => (props?: ImageProps) =>
-      (
-        <CollectionImage
-          collection={item}
-          size={SquareSizes.SIZE_480_BY_480}
-          {...props}
-        />
-      ),
-    []
-  )
+  const collectionsToAddTo = useSelector(selectCollectionsToAddTo)
 
-  const filteredCollections = useMemo(() => {
-    return ((isAlbumType ? account?.albums : account?.playlists) ?? []).filter(
-      (collection: Collection) =>
-        // Don't allow adding to this collection if already on this collection's page.
-        collection.playlist_id !== currentCollectionId &&
-        collection.playlist_owner_id === account?.user_id
-    )
-  }, [
-    isAlbumType,
-    account?.albums,
-    account?.playlists,
-    account?.user_id,
-    currentCollectionId
-  ])
+  const filteredCollectionsToAddTo = useMemo(() => {
+    return filter
+      ? fuzzySearch(
```

Comment:
remind me, this is not included in the selector because it would get calculated any time the selected data changes instead of on component render?

Reply:
it probably could be in the selector too, but since the desktop version just does local state i decided to follow suit